task/build-paketo-ubi-builder/0.1/build-paketo-ubi-builder.yaml

```diff
@@ -0,0 +1,527 @@
+            export IMAGE
+          fi
+          echo "##########################################################################################"
+          echo "### Running syft on the source directory"
+          echo "##########################################################################################"
+          syft dir:"/var/workdir/$SOURCE_CODE_DIR/$CONTEXT" --output cyclonedx-json="/var/workdir/sbom-source.json"
+          
+          echo "##########################################################################################"
+          echo "### Running syft on the image filesystem"
+          echo "##########################################################################################"
+          # syft dir:"$(cat /shared/container_path)" --output cyclonedx-json="/var/workdir/sbom-image.json"
+          syft scan oci-dir:/shared/konflux-final-image -o cyclonedx-json > /var/workdir/sbom-image.json
+        volumeMounts:
+          - mountPath: /shared
+            name: shared
+        workingDir: /var/workdir/source
+      - computeResources:
+          limits:
+            cpu: 200m
+            memory: 512Mi
+          requests:
+            cpu: 100m
+            memory: 256Mi
+        image: quay.io/redhat-appstudio/sbom-utility-scripts-image@sha256:11851ba63f63dfdcf722e47993f41a1f5f31a7a0dc8aa85b810ce2466daf23af
+        name: prepare-sboms
+        script: |
+          #!/bin/bash
+          set -e
+          if [ "${IMAGE_APPEND_PLATFORM}" == "true" ]; then
+            IMAGE="${IMAGE}-${PLATFORM//[^a-zA-Z0-9]/-}"
+            export IMAGE
+          fi
+          
+          echo "##########################################################################################"
+          echo "## Merging contents of sbom-source.json and sbom-image.json into sbom-cyclonedx.json"
+          echo "##########################################################################################"
+          python3 /scripts/merge_syft_sboms.py
+          
+          if [ -f "sbom-cachi2.json" ]; then
+            echo "##########################################################################################"
+            echo "## Merging contents of sbom-cachi2.json into sbom-cyclonedx.json"
+            echo "##########################################################################################"
+            python3 /scripts/merge_cachi2_sboms.py sbom-cachi2.json sbom-cyclonedx.json >sbom-temp.json
+            mv sbom-temp.json sbom-cyclonedx.json
+          fi
+          
+          echo "##########################################################################################"
+          echo "Creating sbom-purl.json"
+          echo "##########################################################################################"
+          python3 /scripts/create_purl_sbom.py
+          
+          # TODO: How can we get for the paketo stuffs: base_images_from_dockerfile
```

Comment:
will be this TODO addressed?

Reply:
I'm not a build konflux expert. I discovered such a new parameter from buildah task `base_images_from_dockerfile`. So the question is: It it something to be done by all the "buidling" tasks processed by konflux ?

Reply:
https://github.com/konflux-ci/build-definitions/blob/d550bc5f4610fe09081d2c177bf4aef64775a66a/task/buildah-remote/0.2/buildah-remote.yaml#L361

https://github.com/konflux-ci/build-definitions/blob/d550bc5f4610fe09081d2c177bf4aef64775a66a/task/buildah-remote/0.2/buildah-remote.yaml#L549

this is how buildah task is getting images from Dockerfile

Reply:
I dunno how pack works, does it use Dockerfile with parent images or multistage dockerfile?

Reply:
https://github.com/konflux-ci/build-definitions/pull/1660/files#diff-1ffbf0bb39f408bedbfcaf34a68c09e005bd54c1d42fd23bb10c38be04dfc061R225

in this line I see that pack uses a base image, so base image must be included in SBOM

Reply:
pack is a buildpack tool that we are using here to build a `builder` image. 

Such a tool will use the following image
```
// https://github.com/paketo-community/builder-ubi-base/blob/main/builder.toml#L74
[stack]
  build-image = "docker.io/paketocommunity/build-ubi-base:0.0.93"
```
to package different folders = buildpacks defined here within the toml file: https://github.com/paketo-community/builder-ubi-base/blob/main/builder.toml#L3-L31

This image that you call "base-image" is used here within the code to populate its SBOM
```
// https://github.com/konflux-ci/build-definitions/blob/01d22af0dbcf13717a424fc9773b57584f19930f/task/build-paketo-builder-oci-ta/0.1/build-paketo-builder-oci-ta.yaml#L332
syft scan oci-dir:/shared/konflux-final-image -o cyclonedx-json > /var/workdir/sbom-image.json
```

So do you expect more than this ? @MartinBasti

Reply:
We have to provide SBOM and link to SBOM into every parent/base image used in build. You are calling syft on the final image, that's not the same.

Reply:
I'm gonna to test it but nevertheless here is the fix: 79ee75dd